### PR TITLE
add support to 2 boards

### PR DESCRIPTION
**Pull request by CircuitART**
Hi please add support to my 2 new boards: circuitart_esp32s2core & circuitart_stm32f4core

thx

### CODE REVIEW

File: ports/esp32s2/boards/circuitart_esp32s2core/board.h
Comment (hathach) [inline ports/esp32s2/boards/circuitart_esp32s2core/board.h:69]: board ID must follow uf2 format https://github.com/microsoft/uf2

File: ports/stm32f4/boards/circuitart_stm32f4core/board.h
Comment (hathach) [inline ports/stm32f4/boards/circuitart_stm32f4core/board.h:67]: board id must follow uf2 format https://github.com/microsoft/uf2

File: .github/workflows/build.yml
Comment (hathach) [inline .github/workflows/build.yml:29]: incorrect abc order

File: ports/esp32s2/boards/circuitart_esp32s2core/board.h
Comment (hathach) [inline ports/esp32s2/boards/circuitart_esp32s2core/board.h:71]: link is not accessible, are you sure it is not a typo

**Review by hathach — CHANGES_REQUESTED** [review]
look good, there is only a couple of requests (check out comment). Also as part of VID/PID requirement, could you tell how you obtain the VID/PID for your boards.

File: ports/esp32s2/boards/circuitart_esp32s2core/board.h
Comment (CircuitART) [reply ports/esp32s2/boards/circuitart_esp32s2core/board.h:71]: The website is not ready yet, should I change to GitHub repo?

**Comment by CircuitART** [comment]
> look good, there is only a couple of requests (check out comment). Also as part of VID/PID requirement, could you tell how you obtain the VID/PID for your boards.

from the repo of pid.codes

**Review by CircuitART — COMMENTED** [review]
board ID has been corrected in both board.h files, also website has this comment now (// TODO update link)
build.yml abc order has been corrected
VID/PID are from pid.codes but the pull requested has not been made yet

File: ports/esp32s2/boards/circuitart_esp32s2core/board.h
Comment (hathach) [reply ports/esp32s2/boards/circuitart_esp32s2core/board.h]: That is OK, I just want to make sure you doesn't make typo.

**Comment by hathach** [comment]
> board ID has been corrected in both board.h files, also website has this comment now (// TODO update link)
> build.yml abc order has been corrected
> VID/PID are from pid.codes but the pull requested has not been made yet

Can you give link to the PR, we will merge once that PR is merged.

**Comment by CircuitART** [comment]
> > board ID has been corrected in both board.h files, also website has this comment now (// TODO update link)
> > build.yml abc order has been corrected
> > VID/PID are from pid.codes but the pull requested has not been made yet
> 
> Can you give link to the PR, we will merge once that PR is merged.

ok thanks I will update the pull request with PID.CODES PR link

**Comment by hathach** [comment]
> > > board ID has been corrected in both board.h files, also website has this comment now (// TODO update link)
> > > build.yml abc order has been corrected
> > > VID/PID are from pid.codes but the pull requested has not been made yet
> > 
> > 
> > Can you give link to the PR, we will merge once that PR is merged.
> 
> ok thanks I will update the pull request with PID.CODES PR link

I mean can you give the link to the PID.CODES PR link here in the comment section, adding it to the code is optional and up to you.

**Pull request closed without merging**
